NuclearData/ceNeutronData/aceDatabase/aceNeutronDatabase_class.f90

```diff
@@ -399,32 +455,116 @@ subroutine updateMacroXSs(self, E, matIdx, rand)
     integer(shortInt)                     :: i, nucIdx
     real(defReal)                         :: dens
 
-    associate (mat => cache_materialCache(matIdx))
-      ! Set new energy
-      mat % E_tot  = E
-      mat % E_tail = E
+    associate(mat      => self % materials(matIdx), &
+              matCache => cache_materialCache(matIdx))
 
       ! Clean current xss
-      call mat % xss % clean()
+      call matCache % xss % clean()
 
-      ! Construct microscopic XSs
-      do i = 1, size(self % materials(matIdx) % nuclides)
-        dens   = self % materials(matIdx) % dens(i)
-        nucIdx = self % materials(matIdx) % nuclides(i)
+      if (mat % useTMS(E)) then
+        call self % updateRelEnMacroXSs(E, matIdx, rand)
 
-        ! Update if needed
-        if (cache_nuclideCache(nucIdx) % E_tail /= E .or. cache_nuclideCache(nucIdx) % E_tot /= E) then
-          call self % updateMicroXSs(E, nucIdx, rand)
-        end if
+      else
 
-        ! Add microscopic XSs
-        call mat % xss % add(cache_nuclideCache(nucIdx) % xss, dens)
-      end do
+        ! Set new energy
+        matCache % E_tot  = E
+        matCache % E_tail = E
+
+        ! Construct microscopic XSs
+        do i = 1, size(mat % nuclides)
+          dens   = mat % dens(i)
+          nucIdx = mat % nuclides(i)
+
+          ! Update if needed
+          if (cache_nuclideCache(nucIdx) % E_tail /= E .or. cache_nuclideCache(nucIdx) % E_tot /= E) then
+            call self % updateMicroXSs(E, nucIdx, rand)
+          end if
+
+          ! Add microscopic XSs
+          call matCache % xss % add(cache_nuclideCache(nucIdx) % xss, dens)
+        end do
+
+      end if
 
     end associate
 
   end subroutine updateMacroXSs
 
+  !!
+  !! Subroutine to update the macroscopic cross sections in a given material
+  !! at given temperature, sampling a relative energy per each nuclide and applying
+  !! a Doppler correction factor
+  !!
+  !! Args:
+  !!   E [in]         -> Incident neutron energy for which the relative energy xss are found
+  !!   matIdx [in]    -> Index of material for which the relative energy xss are found
+  !!
+  subroutine updateRelEnMacroXSs(self, E, matIdx, rand)
```

Comment:
I have a mild preference for the name to be RelE rather than RelEn, just for consistency with other things.

Reply:
I don't like many capitals in a row in a name! Like updateRelEMacroXSs. The 'E' gets lost in there.